scrapper.py

```diff
@@ -26,38 +36,123 @@ class Crawler:
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        class_bs = article_bs.find('div', class_="b-materials-list b-list_infinity")
+        title_bs = class_bs.find_all('p', class_="b-materials-list__title b-materials-list__title_news")
+        for link in title_bs:
+            link = link.find('a')
+            self.urls.append('https://www.m24.ru' + link['href'])
```

Comment:
it is better to use this method to solely extract urls from given soup
leave adding it to the attribute to `find_articles`
also, before adding a new url to the attribute it is necessary to check that such url has not been added before as many sources duplicate urls on their pages

Reply:
done